pkg/piperutils/descriptorUtils.go

```diff
@@ -0,0 +1,210 @@
+package piperutils
+
+import (
+	"bytes"
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"regexp"
+	"strings"
+
+	"github.com/Masterminds/sprig"
+
+	piperCmd "github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/log"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	// SchemeMajorVersion is the versioning scheme based on the major version only
+	SchemeMajorVersion = `{{(split "." (split "-" .Version)._0)._0}}`
+	// SchemeSemanticVersion is the versioning scheme based on the major.minor.micro version
+	SchemeSemanticVersion = `{{(split "-" .Version)._0}}`
+	// SchemeFullVersion is the versioning scheme based on the full version
+	SchemeFullVersion = "{{.Version}}"
+	// NameRegex is used to match the pip descriptor artifact name
+	NameRegex = "(?s)(.*)name=['\"](.*?)['\"](.*)"
+	// VersionRegex is used to match the pip descriptor artifact version
+	VersionRegex = "(?s)(.*)version=['\"](.*?)['\"](.*)"
+	// MethodRegex is used to identify a method within pip descriptor to dynamically load the version from txt file
+	MethodRegex = "(?s)(.*)\\(\\)"
+)
+
+// BuildDescriptor acts as a general purpose accessor to coordinates
+type BuildDescriptor interface {
+	GetVersion() string
+	SetVersion(string)
+}
+
+// MavenDescriptor holds the unique identifier combination for Maven built Java artifacts
+type MavenDescriptor struct {
+	XMLName    xml.Name `xml:"project"`
+	GroupID    string   `xml:"groupId"`
+	ArtifactID string   `xml:"artifactId"`
+	Version    string   `xml:"version"`
+	Packaging  string   `xml:"packaging"`
+}
+
+// GetVersion returns the Maven descriptor version property
+func (desc *MavenDescriptor) GetVersion() string {
+	return desc.Version
+}
+
+// SetVersion sets the Maven descriptor version property
+func (desc *MavenDescriptor) SetVersion(v string) {
+	desc.Version = v
+}
+
+// PipDescriptor holds the unique identifier combination for pip built Python artifacts
+type PipDescriptor struct {
+	GroupID    string
+	ArtifactID string
+	Version    string
+	Packaging  string
+}
+
+// GetVersion returns the Pip descriptor version property
+func (desc *PipDescriptor) GetVersion() string {
+	return desc.Version
+}
+
+// SetVersion sets the Pip descriptor version property
+func (desc *PipDescriptor) SetVersion(v string) {
+	desc.Version = v
+}
+
+// GetMavenCoordinates reads the coordinates from the maven pom.xml descriptor file
+func GetMavenCoordinates(filename string) (*MavenDescriptor, error) {
+	r, _ := regexp.Compile(`\$\{.*?\}`)
+	data, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to read descriptor file %v", filename)
+	}
+	result := &MavenDescriptor{}
+	err = xml.Unmarshal(data, result)
```

Comment:
What about discarding to read from xml and only use `mvn evaluate` to properly resolve the coordinates?

Reply:
@stippi2 it is anyway planned to replace this part of descriptor utils with what we have in @OliverNocon 's versioning.go and uniform how we interact with different build tool specific descriptors in a central place.